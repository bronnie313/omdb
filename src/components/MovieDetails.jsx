import { NavLink, useParams, useNavigate } from 'react-router-dom'
import { Icon } from '@iconify/react'
import useFetchMovie from './useFetchMovie'

const MovieDetails = () => {
  const { imdbID } = useParams()
  const url = `https://www.omdbapi.com/?i=${imdbID}&apikey=2fa5119d`

  const { isLoading, isError, selectedMovie } = useFetchMovie(url)
  const navigate = useNavigate()

  const handleGoBack = () => {
    navigate(-1)
  }

  if (isLoading) {
    return (
      <div style={{ marginTop: '9rem', textAlign: 'center' }}>
        <Icon icon="eos-icons:bubble-loading" fontSize={'2rem'} />
        <div style={{ marginTop: '1rem' }}></div>
        <p>loading</p>
      </div>
    )
  }

  if (isError) {
    return (
      <div style={{ fontSize: '25px', marginTop: '8rem', textAlign: 'center' }}>
        There is an error ...
      </div>
    )
  }

  const { Poster, Title, Plot, Genre } = selectedMovie

  return (
    <div className="container-2">
      <NavLink onClick={handleGoBack}>
        <Icon className="icon" icon="icon-park-solid:back" />
      </NavLink>
      <div className="details">
        <div className="details-wrapper">
          <img src={Poster} alt={Title} />
          <div className="info">
            <h2 style={{ fontFamily: 'cursive' }}>{Title}</h2>
            <p>{Plot}</p>
            <h4>{Genre}</h4>
          </div>
        </div>
      </div>
    </div>
  )
}

export default MovieDetails
