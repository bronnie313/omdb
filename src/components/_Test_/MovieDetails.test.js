import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import MovieDetails from '../MovieDetails'

test('renders MovieDetails correctly', () => {
  const mockMovieData = {
    imdbID: 'tt1234567',
    Poster: 'poster.jpg',
    Title: 'Papillon',
    Plot: 'This is a mock movie plot',
    Genre: 'Comedy',
  }

  const { container } = render(
    <BrowserRouter>
      <MovieDetails />
    </BrowserRouter>
  )

  expect(container.firstChild).toMatchSnapshot()
})
