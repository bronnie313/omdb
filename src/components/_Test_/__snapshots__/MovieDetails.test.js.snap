// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieDetails correctly 1`] = `
<div
  style="margin-top: 9rem; text-align: center;"
>
  <span />
  <div
    style="margin-top: 1rem;"
  />
  <p>
    loading
  </p>
</div>
`;
